api/api-nest/mlopsapi/src/user/user.service.spec.ts

```diff
@@ -0,0 +1,182 @@
+    return this.mockUsers;
+  }
+
+  findOne(id: number): User {
+    const found = this.mockUsers.find((user) => user.id === id);
+    return found;
+  }
+  findOneByName(name: string): User {
+    console.log(this.mockUsers);
+    const found = this.mockUsers.find((user) => user.name == name);
+    console.log(found);
+    return found;
+  }
+
+  save(user: User): User {
+    if (this.findOneByName(user.name)) {
+      throw new ConflictException();
+    }
+
+    return { id: 4, ...user };
+  }
+
+  delete(id: number): User {
+    return this.findOne(id);
+  }
+}
+
+describe('UserService', () => {
+  let service: UserService;
+
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        UserService,
+        {
+          provide: getRepositoryToken(User),
+          useClass: MockUserRepository,
+        },
+      ],
+    }).compile();
+
+    service = module.get<UserService>(UserService);
+  });
+
+  it('should be defined service', async () => {
+    expect(service).toBeDefined();
+  });
+
+  describe('GET /user', () => {
+    describe('success case', () => {
+      let userList: User[];
+
+      beforeAll(async () => {
+        userList = await service.findAll();
+      });
+
+      it('should return all user list', async () => {
+        expect(3).toBe(userList.length);
+      });
+
+      it('should each user has id & age & name', () => {
+        expect(userList[0]).toHaveProperty('id');
+        expect(userList[0]).toHaveProperty('age');
+        expect(userList[0]).toHaveProperty('name');
+      });
+    });
+  });
+
+  describe('GET /user/:id', () => {
+    let findUser: User;
+
+    describe('success case', () => {
+      it('should return a user with property id, age, name ', async () => {
+        findUser = await service.findUserById(1);
+        // console.log(findUser);
+        expect(findUser).toHaveProperty('id');
+        expect(findUser).toHaveProperty('age');
+        expect(findUser).toHaveProperty('name');
+      });
+    });
+
+    describe('failure case', () => {
+      it('should return 400 if user is not found', async () => {
+        expect(service.findUserById(0)).rejects.toThrow(NotFoundException);
+      });
+    });
+  });
+
+  describe('CREATE /user', () => {
+    let createdUser: User;
+
+    const newUser = new CreateUserDto();
+    newUser.age = 30;
+    newUser.name = 'sm5';
+    // const dupUser = { id: 30, name: 'sm1', age: 30 };
+    const dupUser = new CreateUserDto();
+    dupUser.age = 30;
+    dupUser.name = 'sm1';
+    describe('success case', () => {
+      it('should return a created user', async () => {
+        createdUser = await service.createUser(newUser);
+        expect(newUser.name).toEqual(createdUser.name);
+      });
+    });
+
+    describe('failure case', () => {
+      it('should return 409 if user already exists', async () => {
+        expect(service.createUser(dupUser)).rejects.toThrow(HttpException);
```

Comment:
```suggestion
expect(service.createUser(dupUser)).rejects.toThrow(ConflictException);
```
이런 식으로 좀 더 정확한 에러와 매칭시켜주시면 더 좋을 것 같아요